### PR TITLE
Add `SCIMTokenService`

### PR DESCRIPTION
**Pull request by saquibmian**
This service enables the creation, deletion, and retrieval of a new type of token, SCIM tokens. These tokens can be used to authenticate and authorize with the BSR's SCIM 2.0 API.

### CODE REVIEW

File: proto/buf/alpha/registry/v1alpha1/scim_token.proto [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+  // DeleteToken deletes an existing token.
+  //
+  // This method requires authentication.
+  rpc DeleteSCIMToken(DeleteSCIMTokenRequest) returns (DeleteSCIMTokenResponse);
+}
+
+message CreateSCIMTokenRequest {
+  // The time until which the token should be valid.
+  // Must be in the future. May be null for no expiry.
+  google.protobuf.Timestamp expire_time = 2;
```
Comment (cyinma) [inline proto/buf/alpha/registry/v1alpha1/scim_token.proto:44]: should this start from 1?

Comment (saquibmian) [reply proto/buf/alpha/registry/v1alpha1/scim_token.proto:44]: Ah yes, let me fix.

File: proto/buf/alpha/registry/v1alpha1/scim_token.proto [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+message SCIMToken {
+  string id = 1;
+  google.protobuf.Timestamp create_time = 2;
+  google.protobuf.Timestamp expire_time = 3;
+}
+
+// TokenSCIMService is the SCIM Token service.
+service SCIMTokenService {
+  // CreateToken creates a new token suitable for authentication to the SCIM API.
```
Comment (seankimdev) [inline proto/buf/alpha/registry/v1alpha1/scim_token.proto:29]:
```suggestion
  // CreateToken creates a new token suitable for authentication to the SCIM API.
  //
  // This method requires authentication.
```
nit: assuming this requires authentication as well, but looks like it's copied over from the existing token.proto.

**Review by cyinma — APPROVED** [review]
lgtm!

**Review by seankimdev — APPROVED** [review]
Looks good to me!

**Review by seankimdev — APPROVED** [review]

**Pull request merged**